frontend/src/metabase/modes/components/drill/ObjectDetailDrill.ts

```diff
@@ -29,27 +32,50 @@ function getAction({
       return {
         question: () => objectDetailFKDrillQuestion({ question, clicked }),
       };
-    case "zoom":
-      return { action: () => zoomInRow({ objectId }) };
+    case "zoom": {
+      return {
+        action: () =>
+          zoomInRow({
+            objectId,
+            zoomedRowColumnIndex: getZoomedRowColumnIndex(clicked),
+          }),
+      };
+    }
     case "dashboard":
       return { question: () => question };
   }
 }
 
 function getActionExtraData({
   objectId,
+  zoomedRowColumnIndex,
   hasManyPKColumns,
 }: {
   objectId: RowValue;
+  zoomedRowColumnIndex?: number;
   hasManyPKColumns: boolean;
 }) {
   if (!hasManyPKColumns) {
     return {
-      extra: () => ({ objectId }),
+      extra: () => ({ objectId, zoomedRowColumnIndex }),
     };
   }
 }
 
+function getZoomedRowColumnIndex(clicked: ClickObject | undefined) {
+  if (!clicked) {
+    return;
+  }
+
+  const clickedColumnId = clicked?.column?.id;
+
+  // This way to find the index does seem too complicated.
+  // With it we can do drills in tables with reordered/hidden columns.
+  // It ensures we find the correct column position
+  // according to the way result rows/columns are ordered later when the drill happens.
+  return clicked?.data?.findIndex(entry => clickedColumnId === entry.col.id);
```

Comment:
Wdyt about using `column.name` instead of the index here? Relying on an index to find the right is more fragile because it assumes that for instance here in `clicked.data` we always have entries in certain order. It is correct now, but if this changes in the scope of other work, this will silently break. Also, using indices is less descriptive because it can mean two things — column position in data or column position in UI, they can be different

Reply:
Just to make sure I understand: we'd use the column name all the way to the selector, instead of the index?

Or just to match in this find, and then we'd go with an index?

One secondary consequence of this are urls.

Drilling from a question inside a dashboard yields a url like `question/:questionId/:objectId/:columnIndex`, which will look like `question/2-some-slug/2/2` or by default `question/2-some-slug/2` if the columnIndex is 0 as is typical.

If we pass the name all the way the url will have some string as the last param. No big deal but maybe less satisfying I guess.